waspc/data/Generator/templates/react-app/tsconfig.json

```diff
@@ -4,7 +4,9 @@
     // Temporary loosen the type checking until we can address all the errors.
     "jsx": "preserve",
     "allowJs": true,
-    "strict": false
+    "strict": false,
+    // Allow importing pages with the .tsx extension.
+    "allowImportingTsExtensions": true
```

Comment:
Why did we need to add this one?

Reply:
We commented on this IRL, the fact that the router file change from `jsx` to `tsx` made the pattern of defining Wasp pages with `.tsx` incorrect. 

For example, if the user defined `MyPage.tsx` and it got imported by Wasp in `router.tsx`, this wouldn't allow building for production.

We agreed to keep the router as `router.jsx` and just reexport TS types from it (this seems to work just fine) to avoid modifying the `tsconfig.json` if we don't have to. 

We will rethink this in the future when we go for solving the general issue of imports with ts vs. js vs none extensions.